### PR TITLE
Website endpoint

### PR DESCRIPTION
**Pull request by josh-termly**
* Document The website endpoint
* Start documenting errors

### CODE REVIEW

File: endpoints/websites.md
Comment (dkreft-termly) [inline endpoints/websites.md]: "Websites represent a single website and each stores..."

"...will reference a website in some way."

File: endpoints/websites_delete.md
Comment (dkreft-termly) [inline endpoints/websites_delete.md]: "...or invalid, ..."  (comma after "invalid")

"...will be an [error object]..."

File: endpoints/websites_delete.md
Comment (dkreft-termly) [inline endpoints/websites_delete.md:53]: There should be an explicit call-out regarding `_foo` data in general, and `_idx` in particular...probably in the "Overview" section.

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: "At least..."

What will happen if `ids` is empty?

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: "...0 or more objects with this shape:"

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: I'd like to encourage you to sort the keys in each object....same with the bullet list, below.

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md:69]: Should we not also include the data type of each datum?

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: "how"

What are the units? Milliseconds? Seconds? Months? Days? Years?

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: hell? smell? bell? tell? *well*?

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: Should be? Or will be?

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md:87]: Might want to include an `address2`. I'm not sure what kind of addresses need it, but it seems to be pretty much ubiquitous in address forms I've seen.

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md:91]: I'm not sure that the explanatory text is necessary or helpful; the keys are self-documenting.

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: JavaScript :-)

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: "Multiple..."

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md:182]: The more I see of this, the more nervous I get that we're going to run into URL length limitations. This is a very, very verbose way to request this data. :-(

File: endpoints/websites_post.md
Comment (dkreft-termly) [inline endpoints/websites_post.md]: "Once created, the..."

"...request body." (missing period)

File: endpoints/websites_post.md
Comment (dkreft-termly) [inline endpoints/websites_post.md]: TODO?

File: endpoints/websites_post.md
Comment (dkreft-termly) [inline endpoints/websites_post.md]: Why is GitHub making your colon red? (Huh huh heh heh mm mm heh...he said "colon").

File: endpoints/websites_post.md
Comment (dkreft-termly) [inline endpoints/websites_post.md:84]: Add blank line between LL88,89

File: endpoints/websites_post.md
Comment (dkreft-termly) [inline endpoints/websites_post.md]: "Submit multiple websites, one of which...error."

File: endpoints/websites_put.md
Comment (dkreft-termly) [inline endpoints/websites_put.md]: "Once created, ..."

File: endpoints/websites_put.md
Comment (dkreft-termly) [inline endpoints/websites_put.md]: TODO?

File: endpoints/websites_put.md
Comment (dkreft-termly) [inline endpoints/websites_put.md]: Heh heh mm mm heh...colon.

File: endpoints/websites_put.md
Comment (dkreft-termly) [inline endpoints/websites_put.md]: "...or invalid, ..."

File: endpoints/websites_put.md
Comment (dkreft-termly) [inline endpoints/websites_put.md]: Blank line betwixt

File: endpoints/websites_put.md
Comment (dkreft-termly) [inline endpoints/websites_put.md]: "Submit multiple websites, one of which does not exist."

(comma, period)

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: "GET"

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: "accour"??

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: To where? I think you mean "Too." :-)

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: too

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: "Incorrect"

**Review by dkreft-termly — CHANGES_REQUESTED** [review]

File: endpoints/websites_get.md
Comment (ghost) [reply endpoints/websites_get.md:182]: this same thing occurred to me, then I read this: https://stackoverflow.com/questions/812925/what-is-the-maximum-possible-length-of-a-query-string

I think we are asking for trouble regardless, but The Internet At Large seems to be mostly comfortable. Ted and Gino would raise hell I'm guessing...

File: endpoints/websites_delete.md
Comment (josh-termly) [reply endpoints/websites_delete.md:53]: will add to `README.md` as it applies to the entire API

**Review by ghost — COMMENTED** [review]
I'm good with Dan changes too

File: endpoints/websites_get.md
Comment (josh-termly) [reply endpoints/websites_get.md:182]: it is not great, but we are limiting the number of items they can request at once.  So I think we can safely keep it to 2048K (as recommended).  However, I am open to ideas.

File: endpoints/websites_get.md
Comment (josh-termly) [reply endpoints/websites_get.md]: can we hold off sorting until we have the API approved from Access Group and know we are not going to do any major changes?

File: endpoints/websites_get.md
Comment (josh-termly) [reply endpoints/websites_get.md:69]: I believe that they are in lines 33 to 64

File: endpoints/websites_get.md
Comment (josh-termly) [reply endpoints/websites_get.md:91]: These keys are self documenting but not all of the keys in the API are so I would lean to keeping them in for consistency.

File: endpoints/websites_get.md
Comment (josh-termly) [reply endpoints/websites_get.md:87]: I know this is small but I would say anything that is not required we keep out for now to keep the scope as tight as possible considering the deadlines

File: endpoints/websites_post.md
Comment (josh-termly) [reply endpoints/websites_post.md]: represented by the object below.  We could combine the 2 if that is preferred.

File: endpoints/websites_put.md
Comment (josh-termly) [reply endpoints/websites_put.md]: object is below

File: README.md
Comment (dkreft-termly) [inline README.md]: I'd remove the asterisk here...it's not really a part of the bullet list...at least not on the same level as that which follows (the following line, if anything, is a detail reinforcing L10).

File: endpoints/websites.md
Comment (dkreft-termly) [inline endpoints/websites.md]: English is dumb.

"...settings that control..."  (the settings control, it's not the website that controls)

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: Third sentence should start with "If" not "if."

Comment (dkreft-termly) [reply endpoints/websites_get.md]: Also, the last sentence is screaming for an actual example. A picture is worth a thousand words.

File: endpoints/websites_get.md
Comment (dkreft-termly) [reply endpoints/websites_get.md]: I'm not sure why sorting should be contingent upon what AG says. I would think that if you're going to present this to AG, then it should be as readable as possible...but, okay.

File: endpoints/websites_get.md
Comment (dkreft-termly) [reply endpoints/websites_get.md:69]: 🤦

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md]: This is confusing....you say integer, but it's wrapped in quotes. This communicates to me that the server is expecting to receive a string, not an integer.

File: endpoints/websites_get.md
Comment (dkreft-termly) [reply endpoints/websites_get.md]: Unaddressed.

When you say "should," it implies "wiggle room."

File: endpoints/websites_get.md
Comment (dkreft-termly) [reply endpoints/websites_get.md:87]: Easy enough to add later, I suppose.

File: endpoints/websites_get.md
Comment (dkreft-termly) [reply endpoints/websites_get.md]: *cough* Unaddressed. The "s" really should be capitalized.

File: endpoints/websites_get.md
Comment (dkreft-termly) [reply endpoints/websites_get.md:182]: I would prefer a more normalized data structure, but that's difficult if we're going to be allowing requests to span `account_id`s. :-\

File: endpoints/websites_post.md
Comment (dkreft-termly) [reply endpoints/websites_post.md]: I think I would just combine the two:

```
[
  {
    "account_id": "<string>",
    // etc.
  }
]
```

File: endpoints/websites_put.md
Comment (dkreft-termly) [reply endpoints/websites_put.md]: See my response above.

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: Remove...unnecessary and inconsistent.

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: The bold text is confusing. If you're representing an actual value, I'd think it'd be more appropriate to use

```markdown
`"validation_error"`
```

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: ditto

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: Again, I don't think a "Response" subheader is necessary or useful here.

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: Nix. Same throughout this document.

File: error_object.md
Comment (dkreft-termly) [reply error_object.md]: Afterthought...I wonder if this shouldn't be `"validation_errors"` (note the plural) so that there's consistency with the datum that contains the errors?

File: error_object.md
Comment (dkreft-termly) [inline error_object.md:73]: See L51, which states that this key will only exist when `{ error: "validation_error" }`

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: We need to include an `error` not a `message`. This is going to make i18n very difficult.

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: The second sentence should probably say "one" instead of "1."

**Review by dkreft-termly — CHANGES_REQUESTED** [review]

File: endpoints/websites_get.md
Comment (justrudd) [reply endpoints/websites_get.md:182]: We're following the pattern of GraphQL (https://graphql.org/learn/serving-over-http/#get-request). Queries for `GET` are passed in the query string. The main difference being we are using JSON not our own query language.

Comment (justrudd) [reply endpoints/websites_get.md:182]: If we want to come up with our own query language, we can. But it's still going to be encoded and passed in a query string parameter. And there would be no tools for generating it. I picked a format that everyone knows and doesn't require us to write a client.

File: endpoints/websites_get.md
Comment (justrudd) [inline endpoints/websites_get.md]: We `id`. Don't need `uuid` as well.

File: endpoints/websites_get.md
Comment (justrudd) [reply endpoints/websites_get.md]: Probably because of syntax highlighting. If you don't wrap it in quotes, it looks like this.

Screen Shot 2021-09-22 at 12 59 10 PM

But I agree with Dan. I'd rather it be correct. So remove the quotes.

Comment (justrudd) [reply endpoints/websites_get.md]: Or remove the language identifier and just let it be monospace font.

File: endpoints/websites_get.md
Comment (josh-termly) [reply endpoints/websites_get.md]: If you mean encoding of the parameters needs an example each example has the encoded and non encoded parameters

File: endpoints/websites_get.md
Comment (dkreft-termly) [inline endpoints/websites_get.md:18]: What should happen if the array is empty?

File: error_object.md
Comment (dkreft-termly) [inline error_object.md]: "f"?

**Review by dkreft-termly — APPROVED** [review]

**Review by ghost — APPROVED** [review]
LGTM

File: endpoints/websites_get.md
Comment (josh-termly) [reply endpoints/websites_get.md:18]: added example

**Pull request merged**
